playbooks/roles/dev-patcher/tasks/main.yml

```diff
@@ -29,6 +29,17 @@
 
 - name: Get patch details
   include_tasks: patch-repos.yml
-  loop: "{{ dev_patcher_patches }}"
+  loop: "{{ dev_patcher_patches | default([], True) }}"
   loop_control:
     loop_var: patchnumber
+
+- name: Check if there is local code change to cloned repos on deployer
+  stat:
+    path: "{{ role_path }}/files/repos/"
```

Comment:
is role_path/files a proper path to store this? We can possible have repos inside the roles???

Seems a bit awkward to me, this should point to an external, different dir which is out of the way. We already have a dir for things out of the roles scope/external and that is `"{{ playbook_dir }}/../files/`

Can you point this there instead as to be in sync?

Reply:
This is dev-only functionality and we are NOT likely to check-in temporary code changes to branch. Hopefully, we will have a upstream review after successful local testing out of that.

The reason this is in dev-patcher as its only needed during dev-work and does not have to visible externally. 
 

This is kind of file structure I have to use local-only changes

arun:~/myFolder/myWork/suse_cloud/arun_forks/socok8s/playbooks/roles/dev-patcher/files$ tree repos
repos
├── airship-deckhand
│   ├── images
│   │   └── deckhand
│   │       ├── Dockerfile.opensuse
│   │       └── Dockerfile.ubuntu
│   └── Makefile
├── airship-shipyard
│   ├── charts
│   │   └── shipyard
│   │       └── templates
│   ├── images
│   │   └── airflow
│   └── src
│       └── bin
│           └── shipyard_airflow
│               └── entrypoint.sh
└── openstack-helm
    └── keystone
        └── templates
            └── bin